northbound/rest/src/main/java/org/eclipse/sensinact/northbound/rest/impl/RestNorthbound.java

```diff
@@ -193,6 +196,15 @@ public AbstractResultDTO resourceGet(final String providerId, final String servi
         return handleQuery(query);
     }
 
+    @Override
+    public AbstractResultDTO getSnapshot(final boolean includeMetadata, List<ResourceSelector> filter) {
+        final QuerySnapshotDTO query = new QuerySnapshotDTO();
+        query.uri = new SensinactPath();
+        query.filter = filter;
+        query.includeMetadata = Optional.ofNullable(includeMetadata).orElse(false);
```

Comment:
A primitive boolean can never be `null`

```suggestion
        query.includeMetadata = includeMetadata;
```